src/providers/FileSystemProvider/TextSearchProvider.ts

```diff
@@ -104,13 +105,94 @@ export class TextSearchProvider implements vscode.TextSearchProvider {
     } else {
       const sysStr = params.has("system") && params.get("system").length ? params.get("system") : "0";
       const genStr = params.has("generated") && params.get("generated").length ? params.get("generated") : "0";
+
+      let uri = options.folder;
+
+      if (!params.get("filter")) {
+        // Unless isfs spec already includes a non-empty filter (which it rarely does), apply includes and excludes at the server side.
+        // Convert **/ separators and /** suffix into multiple *-patterns that simulate these elements of glob syntax.
+
+        // Function to convert glob-style filters into ones that the server understands
+        const convertFilters = (filters: string[]): string[] => {
+          // Use map to prevent duplicates in final result
+          const filterMap = new Map<string, void>();
+
+          // The recursive function we use
+          const recurse = (value: string): void => {
+            const parts = value.split("**/");
+            if (parts.length < 2) {
+              // No more recursion
+              if (value.endsWith("/**")) {
+                filterMap.set(value.slice(0, -1));
+                filterMap.set(value.slice(0, -3));
+              } else {
+                filterMap.set(value);
+              }
+            } else {
+              const first = parts[0];
+              const rest = parts.slice(1);
+              recurse(first + "*/" + rest.join("**/"));
+              recurse(first + rest.join("**/"));
+            }
+          };
+
+          // Invoke our recursive function
+          filters
+            .filter((value) => csp || !value.match(/\.([a-z]+|\*)\/\*\*$/)) // drop superfluous entries ending .xyz/** or .*/** when not handling CSP files
+            .forEach((value) => {
+              recurse(value);
+            });
+
+          // Convert map to array and return it
+          const results: string[] = [];
+          filterMap.forEach((_v, key) => {
+            results.push(key);
+          });
+          return results;
+        };
+
+        // Function to get one of the two kinds of exclude settings as an array
+        const getConfigExcludes = (key: string) => {
+          return Object.entries(vscode.workspace.getConfiguration(key, options.folder).get("exclude"))
```

Comment:
Is this call to `getConfiguration()` necessary? There's an`options.useIgnoreFiles`,`options.useGlobalIgnoreFiles` and `options.useParentIgnoreFiles` but I'm not sure if they're the same thing

Reply:
Yes it's necessary. The three options you mention are only booleans, designed to tell the searchprovider whether or not it should also apply the contents of those files. We're not currently doing anything with them, but I don't see that as a problem at the moment.